tests/build-and-run-tests.sh

```diff
@@ -28,14 +28,23 @@ function ciSkip {
     exit 0
 }
 
+# Finds the branching point of two commits. 
+# For example, let B and D be two commits, and their ancestry graph as A -> B, A -> C -> D. 
+# Given commits B and D, it returns A. 
+function getLowestCommonAncestorCommitHash {
```

Comment:
There are other cases where this function will be used to detect the LCA, that's why I think `getLCA` would be confusing. How about `getBranchingPoint`?